File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:

        existed=set()
        def helper(n, sum):
            while(n>=10):
                sum+=(n%10)**2
                n=n//10
                
                # print(n, sum)
                
            sum+=n**2
            # print(sum)
            
            if(sum in existed):
                return 0
            if(sum==1):
                return 1
            existed.add(sum)
            return helper(sum, 0)
        
        return helper(n, 0)
    
            

File: 0202-happy-number/0202-happy-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function(n) {
        let existed=new Set()
        function helper(n, sum){
            while(n>=10){
                sum+=Math.pow((n%10),2)
                n=Math.floor(n/10)
                
                // console.log(n, sum)
            }    
            sum+=Math.pow(n, 2)
            // console.log(sum)
            
            if(existed.has(sum)) return 0
            if(sum===1) return 1
            
            existed.add(sum)
            return helper(sum, 0)
        }
        return helper(n, 0)
    
            
};